contracts/governance/Staking/Staking.sol

```diff
@@ -298,21 +298,120 @@ contract Staking is
+    function governanceDirectWithdrawVesting(
+        address vesting,
+        address receiver,
+        uint256 startFrom
+    ) external onlyAuthorized whenNotFrozen {
+        _governanceDirectWithdrawVesting(vesting, receiver, startFrom);
+    }
+
+    /**
+     * @notice Withdraws tokens from the staking contract and forwards them
+     * to an address specified by the token owner. Low level function.
+     * @dev Once here the caller permission is taken for granted.
+     * @param _vesting The vesting address.
+     * @param _receiver The receiving address.
+     * @param _startFrom The start value for the iterations.
+     * or just unlocked tokens (false).
+     * */
+    function _governanceDirectWithdrawVesting(
+        address _vesting,
+        address _receiver,
+        uint256 _startFrom
+    ) private {
+        require(_receiver != address(0), "receiver address invalid");
+
+        ITeamVesting teamVesting = ITeamVesting(_vesting);
+
+        VestingConfig memory vestingConfig =
+            VestingConfig(
+                _vesting,
+                teamVesting.startDate(),
+                teamVesting.endDate(),
+                teamVesting.cliff(),
+                teamVesting.duration(),
+                teamVesting.tokenOwner()
+            );
+
+        uint96 tempStake;
+
+        /// @dev Usually we just need to iterate over the possible dates until now.
+        uint256 end;
+
+        uint256 defaultStart = vestingConfig.startDate + vestingConfig.cliff;
+
+        _startFrom = _startFrom >= defaultStart ? _startFrom : defaultStart;
+
+        /// @dev In the unlikely case that all tokens have been unlocked early,
+        ///   allow to withdraw all of them.
+        end = vestingConfig.endDate;
+
+        uint256 totalIterationValue =
+            (_startFrom + (FOUR_WEEKS * getMaxVestingWithdrawIterations()));
+        uint256 adjustedEnd = end < totalIterationValue ? end : totalIterationValue;
+
+        /// @dev Withdraw for each unlocked position.
+        /// @dev Don't change FOUR_WEEKS to TWO_WEEKS, a lot of vestings already deployed with FOUR_WEEKS
+        ///		workaround found, but it doesn't work with TWO_WEEKS
+        for (uint256 i = _startFrom; i < adjustedEnd; i += FOUR_WEEKS) {
+            /// @dev Read amount to withdraw.
+            tempStake = _getPriorUserStakeByDate(_vesting, i, block.number - 1);
+
+            // We need to execute _withdrawNext(until, receiver, false); even if stake for _withdraw(amount, until, receiver, false); is zero.
+            if (tempStake == 0) {
+                tempStake = 1;
+            }
+
+            _governanceWithdrawDirect(tempStake, i, _receiver, vestingConfig);
+        }
+
+        if (adjustedEnd < end) {
+            emit IncompleteGovernanceWithdrawTokens(
+                msg.sender,
+                _receiver,
+                adjustedEnd - FOUR_WEEKS
+            );
+        } else {
+            emit TokensWithdrawn(msg.sender, _receiver);
```

Comment:
pls rename to `GovernanceWithdrawVestedTokens`

Reply:
Fixed in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/447/commits/c4c43c22e2dc30a936960e01e9a8acd6e2053616)